container/lxd/lxd.go

```diff
@@ -27,6 +27,23 @@ var (
 
 const lxdDefaultProfileName = "default"
 
+// ApplyConstraints takes any supported constraints in cons and sets their
+// respective values in the given lxd metadata map.
+func ApplyConstraints(metadata map[string]string, cons constraints.Value) {
+	// apply any constraints
+	if cons.CpuCores != nil {
+		metadata["limits.cpu"] = fmt.Sprintf("%v", *cons.CpuCores)
+	}
+
+	if cons.CpuPower != nil {
+		metadata["limits.cpu.allowance"] = fmt.Sprintf("%v", *cons.CpuPower)
```

Comment:
We don't have a concept of cpu priority in our constraints.  These are the ones that match what constraints are common across the clouds, aside from root-disk, which is an entirely different ball of wax, so we're leaving that one for later.